web-ui/src/components/kudos_card/KudosCard.jsx

```diff
@@ -0,0 +1,278 @@
+import React, { useCallback, useContext, useState } from "react";
+import PropTypes from "prop-types";
+import {
+  Paper,
+  Collapse,
+  Divider,
+  Typography,
+  Avatar,
+  Chip,
+  Button,
+  AvatarGroup,
+  Tooltip,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  TextField,
+} from "@mui/material";
+import { selectCsrfToken, selectProfile } from "../../context/selectors";
+import { AppContext } from "../../context/AppContext";
+import { getAvatarURL } from "../../api/api";
+import DateFnsUtils from "@date-io/date-fns";
+import CheckIcon from "@mui/icons-material/Check";
+import CloseIcon from "@mui/icons-material/Close";
+import TeamIcon from "@mui/icons-material/Groups";
+
+import "./KudosCard.css";
+import { approveKudos, deleteKudos } from "../../api/kudos";
+import { UPDATE_TOAST } from "../../context/actions";
+
+const dateUtils = new DateFnsUtils();
+
+const propTypes = {
+  kudos: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    message: PropTypes.string.isRequired,
+    senderId: PropTypes.string.isRequired,
+    recipientTeam: PropTypes.object,
+    dateCreated: PropTypes.array.isRequired,
+    dateApproved: PropTypes.array,
+    recipientMembers: PropTypes.array,
+  }).isRequired,
+  includeActions: PropTypes.bool,
+  onKudosAction: PropTypes.func,
+};
+
+const KudosCard = ({ kudos, includeActions, onKudosAction }) => {
+  const { state, dispatch } = useContext(AppContext);
+  const csrf = selectCsrfToken(state);
+
+  const [expanded, setExpanded] = useState(true);
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  // const [deleteReason, setDeleteReason] = useState(""); // TODO: setup optional reason for deleting a kudos
```

Comment:
Do we want to implement this now or save it for later?

Reply:
Later is fine for this. I'm not totally sure of the use case. I'll think about this.